urllib3/response.py

```diff
@@ -319,6 +319,8 @@ def from_httplib(ResponseCls, r, **response_kw):
         with ``original_response=r``.
         """
         headers = r.msg
+        validate_headers(headers)
```

Comment:
Note we already have assert_fingerprints which raises on failure. I think that's a good pattern to follow here. :)